src/vanilla/shallow.ts

```diff
@@ -1,25 +1,30 @@
-const isIterable = (obj: object): obj is Iterable<unknown> =>
+type IterableLike<T> = Iterable<T> & {
+  entries: () => IterableIterator<[number, T]>
+}
+
+const isIterable = (obj: object): obj is IterableLike<unknown> =>
   Symbol.iterator in obj
```

Comment:
This code doesn't feel correct. How can we ensure it has entries `() => IterableIterator<[number, T]>`?

Reply:
@dai-shi all `Map`, `Set`, and `Array` have `entries()`. Actually, all iterables have `entries()` so checking `Symbol.iterator` should be enough.